### ISSUE TITLE
Incorrect filtering of last month in Vertica

### ISSUE THREAD

**Issue by sergekzhr** [metabase/metabase#28840]
**Describe the bug**
The Last Month filter returns two months data

**To Reproduce**
Steps to reproduce the behavior (if you can reproduce the bug using the Sample Database, we will find the issue faster):
1. Go to the Editor
2. Add Date filter
3. Last Month
4. Visualize
5. February and January are showing

**Expected behavior**
Today is 2nd March
I expect to see only February

**Additional context**
The filter generated by Metabase:
`"source"."report_date" >= date_trunc('month', (CAST(now() AS timestamp) + (INTERVAL '-1 month')))
   AND "source"."report_date" < date_trunc('month', now())`

Bug is here: 
`SELECT (CAST(now() AS timestamp) + (INTERVAL '-1 month'));`
Returns: `2023-01-31 14:02:12.205931`

Correct way to deduct 1 month in Vertica:
`ADD_MONTHS(report_date, -1)`

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.3+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "17.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.3+7",
    "os.name": "Linux",
    "os.version": "4.18.0-372.19.1.el8_6.x86_64",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "clickhouse",
      "postgres",
      "vertica"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "10.20"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-11-01",
      "tag": "v0.44.6",
      "branch": "release-x.44.x",
      "hash": "5c6ae21"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Comment by camsaul** [comment]
Bot sure when this changed but currently we're generating

```
WHERE
  (
    "public"."test_data_orders"."created_at" >= DATE_TRUNC(
      'month',
      TIMESTAMPADD("month", -1, CURRENT_TIMESTAMP(6))
    )
  )
  AND (
    "public"."test_data_orders"."created_at" < DATE_TRUNC('month', CURRENT_TIMESTAMP(6))
  )
  ```

**Comment by camsaul** [comment]
```sql
SELECT
  date_trunc('month', (CAST(now() AS timestamp) + (INTERVAL '-1 month'))) AS old_start,
  date_trunc('month', now()) AS old_end,
  date_trunc('month', timestampadd('month', -1, current_timestamp(6))) AS new_start,
  date_trunc('month', current_timestamp(6)) AS new_end
```

gives me 

```clj
{:cols
 [{:name "old_start", :base_type :type/DateTime}
  {:name "old_end", :base_type :type/DateTimeWithLocalTZ}
  {:name "new_start", :base_type :type/DateTimeWithLocalTZ}
  {:name "new_end", :base_type :type/DateTimeWithLocalTZ}],
 :rows [[#t "2024-03-01T00:00" #t "2024-04-01T07:00Z[UTC]" #t "2024-03-01T08:00Z[UTC]" #t "2024-04-01T07:00Z[UTC]"]]}
```

so it seems pretty clear the old issues were related to the old start being a `timestamp` rather than `timestamp with time zone`. This was fixed in Metabase 46 by #26548

Fixed by #26548

**Comment by camsaul** [comment]
Fixed by https://github.com/metabase/metabase/pull/26548
